documentation/binding-data/tutorial-flow-data-provider.asciidoc

```diff
@@ -272,28 +272,45 @@ The lazy loaded binding in `ComboBox` is always filtered by the string typed
 in by the end user.
 Initially, when there is no filter input yet, the filter is an empty string.
 
-// TODO change this into data view style once the changes are in ComboBox
-
-`ComboBox` currently uses the old data provider naming convention.
-Also, the count of total matches is currently required.
+*The `ComboBox` examples listed below use an API available only in Vaadin 18+.*
 
 For example, you can handle filterable lazy data binding to a Spring Data
 repository as follows:
 
 [source, java]
 ----
 ComboBox<Person> cb = new ComboBox<>();
-cb.setDataProvider((String filter, int offset, int limit) -> {
-    return repo.findByNameLikeIgnoreCase(
-            "%" + filter + "%", // <1>
-            PageRequest.of(offset / limit, limit)
-    ).stream();
-}, filter -> {
-    return (int) repo.countByNameLikeIgnoreCase("%" + filter + "%"); // <2>
-});
+cb.setItems(
+        query -> repo.findByNameLikeIgnoreCase( // <1>
+                "%" + query.getFilter().orElse("") + "%", // <2>
+                PageRequest.of(query.getPage(), query.getPageSize()))
+                .stream()
+);
 ----
-<1> Add `%` marks to filter for an SQL "LIKE" query.
-<2> The total amount of items matching the filter is required.
+<1> Here we use only the fetch callback for lazy loading with undefined item count. The item count callback is optional.
```

Comment:
I would stop introducing more content with the "we" and "you" format as it seems to be discouraged.

I'm not sure if the // <1> here really makes sense and matches the code that well. Would be odd that it describes something that is not visible in the code and would not even be on this line.
Instead I'd try to add separately something like

"The above example uses a fetch callback for lazy loading items, and the combo box will fetch more items as the user scrolls the dropdown, until there are no more items returned. In case it is desired to have the dropdown's scrollbar reflect the exact number of items matching the filter, an optional item count callback can be used As shown in the example below: ..."

Reply:
done